src/plugins/action/action_impl.cpp

```diff
@@ -349,6 +349,10 @@ void ActionImpl::takeoff_async(const Action::ResultCallback& callback) const
     command.command = MAV_CMD_NAV_TAKEOFF;
     command.target_component_id = _parent->get_autopilot_id();
 
+#ifdef ARDUPILOT
+    command.params.param7 = 1.0; // Ardupilot doesn't take off without altitude set
```

Comment:
@julianoes: does PX4 use this param? If yes we could move to it instead of the px4-specific parameter, right?

Reply:
It does and I would like to switch to it but guess what: PX4 wants AMSL! So this is something we need to fix but for now this workaround make sense.

Although, hang on. @levevg we probably should save the takeoff altitude here:
https://github.com/mavlink/MAVSDK/blob/3ef7dc3040cdb14297623e44a6f6ac42479ee326/src/plugins/action/action_impl.cpp#L583-L586
and then use it.